cli/main.rs

```diff
@@ -355,7 +356,13 @@ async fn lint_command(
     return Ok(());
   }
 
-  lint::lint_files(files).await
+  let reporter = if json {
+    lint::LintReporterKind::Json
+  } else {
+    lint::LintReporterKind::Pretty
+  };
+
+  lint::lint_files(files, reporter).await
```

Comment:
I guess we should pass `json` argument to `lint_files` - no need to leak `LintReporterKind` outside of `lint` module